src/Raven.Server/Documents/ETL/Providers/Queue/AwsSqs/AwsSqsEtl.cs

```diff
@@ -0,0 +1,270 @@
+            }
+
+            if (Configuration.SkipAutomaticQueueDeclaration == false &&
+                _alreadyCreatedQueues.ContainsKey(queueName) == false)
+                AsyncHelpers.RunSync(() => CreateQueue(_queueClient, queueName));
+
+            var batchMessages = new List<SendMessageBatchRequestEntry>();
+
+            foreach (AwsSqsItem queueItem in queue.Items)
+            {
+                CancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    var sendMessageEntry = new SendMessageBatchRequestEntry
+                    {
+                        Id = queueItem.DocumentId,
+                        MessageBody = SerializeCloudEvent(queueItem)
+                    };
+
+                    batchMessages.Add(sendMessageEntry);
+
+                    if (batchMessages.Count == 10)
+                    {
+                        SendBatchMessages(queueName, batchMessages, queue, idsToDelete, tooLargeDocsErrors);
+                        count += batchMessages.Count;
+                        batchMessages.Clear();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new QueueLoadException($"Failed to prepare message, error reason: '{ex.Message}'", ex);
+                }
+            }
+
+            if (batchMessages.Count > 0)
+            {
+                SendBatchMessages(queueName, batchMessages, queue, idsToDelete, tooLargeDocsErrors);
+                count += batchMessages.Count;
+                batchMessages.Clear();
+            }
+
+            if (tooLargeDocsErrors.Count > 0)
+            {
+                Database.NotificationCenter.EtlNotifications.AddLoadErrors(Tag, Name, tooLargeDocsErrors,
+                    "ETL has partially loaded the data. " +
+                    "Some of the documents were too big (>256KB) to be handled by Aws Sqs. " +
+                    "It caused load errors, that have been skipped. ");
+            }
+        }
+
+        return count;
+    }
+
+    private void SendBatchMessages(string queueName, List<SendMessageBatchRequestEntry> batchMessages,
+        QueueWithItems<AwsSqsItem> queue, List<string> idsToDelete, Queue<EtlErrorInfo> tooLargeDocsErrors)
+    {
+        try
+        {
+            var sendMessageBatchRequest = new SendMessageBatchRequest
+            {
+                QueueUrl = _alreadyCreatedQueues.GetValueOrDefault(queueName),
+                Entries = batchMessages
+            };
+
+            AsyncHelpers.RunSync(() => _queueClient.SendMessageBatchAsync(sendMessageBatchRequest));
+
+            if (queue.DeleteProcessedDocuments)
+            {
+                idsToDelete.AddRange(batchMessages.Select(entry => entry.Id));
+            }
+        }
+        catch (Exception)
+        {
+            // Retry sending one by one if batch fails
+            foreach (var message in batchMessages)
```

Comment:
I'd rather recommend changing  `void SendBatchMessages()` to `bool TrySendBatchMessages()` and have the code sending the messages one by one outside, in a dedicated method, instead of putting it in `catch` block.

I think that here we might want to log the fact of getting an exception (Warning level?).

Reply:
fixed